### DISCUSSION TITLE
What is the most straight forward way to re-number my active workspaces?

### DISCUSSION THREAD

**Discussion by ChangelingX2** [i3/i3#5481] [Q&A] [answered]
Hi all, I am really just looking to be pointed to the right documentation, but for the life of me I can´t find what I need. I am sure this is a basic thing but I honestly can´t find it anywhere.

Backstory is, I have a finnicky monitor and once or twice a month I have to fight with it to get it to actually register as a monitor and display a desktop again. Inevitably, I also always end up with some extra workspaces stacked onto my main monitor at the end of the process. I can delete these, but it leaves me with weird workspace numbers remaining, e.g. 1, 2, 4 rather than 1, 2, 3.

Is there a simple way to either 

A: change the number on the mis-numbered workspace.

B: "collapse" the current workspace numbering so that it doesn't have any gaps in the numbering?

Like I said, I am sure there's clear documentation on this but I cannot for the life of me find it. I have found scripts that require the API extension to be installed but there has to be a simpler way to accomplish it. I just want to re-number a window! *sobs*

**Comment by orestisfl** [comment, marked as answer]
Are you looking for the `rename workspace` command? https://i3wm.org/docs/userguide.html#_renaming_workspaces

**Reply by ChangelingX2** [reply]
Yeah. My girlfriend showed me an alternative way to do it (binding moving workspaces between outputs to a pair of keys) but that also was helpful. I tested it and I was able to do i3-msg 'rename 1 to 4' (and back) to test and it worked fine. Thanks.
